### PR TITLE
[ISSUE #1709] Add test case for shenyu-admin module

### LINKED ISSUE #1709: Add test case for shenyu-admin module
## Task

A clear and concise description of Task

Task List:

- [ ] task1
- [ ] task2
- [ ] task3

Please read the [Contribution Guideline](https://shenyu.apache.org/community/contributor/) before submitting the PR

### PR DESCRIPTION
**Pull request by renzhuyan**
// Describe your PR here; eg. Fixes #1709 

Make sure that:

- [ ] You have read the [contribution guidelines](https://shenyu.apache.org/community/contributor/).
- [ ] You submit test cases (unit or integration tests) that back your changes.
- [ ] Your local test passed `mvn clean install -Dmaven.javadoc.skip=true`.

### CODE REVIEW

File: shenyu-admin/src/test/java/org/apache/shenyu/admin/listener/etcd/EtcdClientTest.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    @Test
+    public void close() {
+        etcdClient.close();
+    }
+
+    @Test
+    public void put() {
+        CompletableFuture<PutResponse> put = mock(CompletableFuture.class);
+        when(client.getKVClient().put(ByteSequence.from(TEST_KEY, StandardCharsets.UTF_8), ByteSequence.from(TEST_VALUE, StandardCharsets.UTF_8))).thenReturn(put);
+        etcdClient.put(TEST_KEY, TEST_VALUE);
```
Comment (dengliming) [inline shenyu-admin/src/test/java/org/apache/shenyu/admin/listener/etcd/EtcdClientTest.java:70]: It also makes sense to verify what we expect. not just calling. : )
`verify(client.getKVClient(), times(1)).put(any(ByteSequence.class), any(ByteSequence.class));`

File: shenyu-admin/src/test/java/org/apache/shenyu/admin/listener/etcd/EtcdClientTest.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    }
+
+    @Test
+    public void delete() {
+        CompletableFuture<DeleteResponse> delete = mock(CompletableFuture.class);
+        when(client.getKVClient().delete(ByteSequence.from(TEST_KEY, StandardCharsets.UTF_8))).thenReturn(delete);
+        etcdClient.delete(TEST_KEY);
```
Comment (dengliming) [inline shenyu-admin/src/test/java/org/apache/shenyu/admin/listener/etcd/EtcdClientTest.java:77]: The same as above.

File: shenyu-admin/src/test/java/org/apache/shenyu/admin/listener/etcd/EtcdClientTest.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    }
+
+    @Test
+    public void deleteEtcdPathRecursive() {
+        etcdClient.deleteEtcdPathRecursive(TEST_KEY);
```
Comment (dengliming) [inline shenyu-admin/src/test/java/org/apache/shenyu/admin/listener/etcd/EtcdClientTest.java:82]: The same as above.

File: shenyu-admin/src/test/java/org/apache/shenyu/admin/listener/etcd/EtcdDataDataChangedListenerTest.java [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+
+    /**
+     * test case onAppAuthChanged event.
+     */
+    @Test
+    public void testOnAppAuthChanged() {
+        AppAuthData appAuthData = AppAuthData.builder().appKey(MOCK_APP_KEY).appSecret(MOCK_APP_SECRET).build();
+        etcdDataDataChangedListener.onAppAuthChanged(ImmutableList.of(appAuthData), DataEventTypeEnum.CREATE);
+        etcdDataDataChangedListener.onAppAuthChanged(ImmutableList.of(appAuthData), DataEventTypeEnum.UPDATE);
+        etcdDataDataChangedListener.onAppAuthChanged(ImmutableList.of(appAuthData), DataEventTypeEnum.DELETE);
```
Comment (dengliming) [inline shenyu-admin/src/test/java/org/apache/shenyu/admin/listener/etcd/EtcdDataDataChangedListenerTest.java:72]:
```
verify(etcdClient, times(2)).put(any(String.class), any(String.class));
verify(etcdClient, times(1)).delete(any(String.class));
```

File: shenyu-admin/src/test/java/org/apache/shenyu/admin/listener/etcd/EtcdDataDataChangedListenerTest.java [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+    /**
+     * test case onMetaDataChanged event.
+     */
+    @SneakyThrows
+    @Test
+    public void testOnMetaDataChanged() {
+        MetaData metaData = MetaData.builder().id(MOCK_ID).path(MOCK_PATH).appName(MOCK_APP_NAME).build();
+        etcdDataDataChangedListener.onMetaDataChanged(ImmutableList.of(metaData), DataEventTypeEnum.CREATE);
+        etcdDataDataChangedListener.onMetaDataChanged(ImmutableList.of(metaData), DataEventTypeEnum.UPDATE);
+        etcdDataDataChangedListener.onMetaDataChanged(ImmutableList.of(metaData), DataEventTypeEnum.DELETE);
```
Comment (dengliming) [inline shenyu-admin/src/test/java/org/apache/shenyu/admin/listener/etcd/EtcdDataDataChangedListenerTest.java:84]: The same as above.

File: shenyu-admin/src/test/java/org/apache/shenyu/admin/listener/etcd/EtcdDataDataChangedListenerTest.java [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+
+    /**
+     * test case onPluginChanged event.
+     */
+    @Test
+    public void testOnPluginChanged() {
+        PluginData pluginData = PluginData.builder().id(MOCK_ID).name(MOCK_NAME).config(MOCK_CONFIG).build();
+        etcdDataDataChangedListener.onPluginChanged(ImmutableList.of(pluginData), DataEventTypeEnum.CREATE);
+        etcdDataDataChangedListener.onPluginChanged(ImmutableList.of(pluginData), DataEventTypeEnum.UPDATE);
+        etcdDataDataChangedListener.onPluginChanged(ImmutableList.of(pluginData), DataEventTypeEnum.DELETE);
```
Comment (dengliming) [inline shenyu-admin/src/test/java/org/apache/shenyu/admin/listener/etcd/EtcdDataDataChangedListenerTest.java:95]: The same as above.

File: shenyu-admin/src/test/java/org/apache/shenyu/admin/listener/etcd/EtcdDataDataChangedListenerTest.java [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+     * test case onSelectorChanged event.
+     */
+    @Test
+    public void testOnSelectorChanged() {
+        SelectorData selectorData = SelectorData.builder().id(MOCK_ID).name(MOCK_NAME).pluginName(MOCK_PLUGIN_NAME).build();
+
+        etcdDataDataChangedListener.onSelectorChanged(ImmutableList.of(selectorData), DataEventTypeEnum.CREATE);
+        etcdDataDataChangedListener.onSelectorChanged(ImmutableList.of(selectorData), DataEventTypeEnum.UPDATE);
+        etcdDataDataChangedListener.onSelectorChanged(ImmutableList.of(selectorData), DataEventTypeEnum.REFRESH);
+        etcdDataDataChangedListener.onSelectorChanged(ImmutableList.of(selectorData), DataEventTypeEnum.DELETE);
```
Comment (dengliming) [inline shenyu-admin/src/test/java/org/apache/shenyu/admin/listener/etcd/EtcdDataDataChangedListenerTest.java:108]: The same as above.

File: shenyu-admin/src/test/java/org/apache/shenyu/admin/listener/etcd/EtcdDataDataChangedListenerTest.java [hunk 8]
```diff
@@ -0,0 +1,128 @@
...
+                .id(MOCK_ID)
+                .name(MOCK_NAME)
+                .pluginName(MOCK_PLUGIN_NAME)
+                .selectorId(MOCK_SELECTOR_ID)
+                .build();
+
+        etcdDataDataChangedListener.onRuleChanged(ImmutableList.of(ruleData), DataEventTypeEnum.CREATE);
+        etcdDataDataChangedListener.onRuleChanged(ImmutableList.of(ruleData), DataEventTypeEnum.UPDATE);
+        etcdDataDataChangedListener.onRuleChanged(ImmutableList.of(ruleData), DataEventTypeEnum.REFRESH);
+        etcdDataDataChangedListener.onRuleChanged(ImmutableList.of(ruleData), DataEventTypeEnum.DELETE);
```
Comment (dengliming) [inline shenyu-admin/src/test/java/org/apache/shenyu/admin/listener/etcd/EtcdDataDataChangedListenerTest.java:126]: The same as above.

File: shenyu-admin/src/test/java/org/apache/shenyu/admin/listener/nacos/NacosDataChangedListenerTest.java [hunk 9]
```diff
@@ -0,0 +1,177 @@
...
+        String config = "{\"divide\":{\"appKey\":\"appKey\",\"appSecret\":\"appSecret\",\"open\":true}}";
+        AppAuthData appAuthData = AppAuthData.builder().appKey(MOCK_APP_KEY).appSecret(MOCK_APP_SECRET).build();
+
+        when(configService.getConfig(NacosPathConstants.AUTH_DATA_ID, NacosPathConstants.GROUP,
+                NacosPathConstants.DEFAULT_TIME_OUT)).thenReturn(config);
+        nacosDataChangedListener.onAppAuthChanged(ImmutableList.of(appAuthData), DataEventTypeEnum.DELETE);
+        nacosDataChangedListener.onAppAuthChanged(ImmutableList.of(appAuthData), DataEventTypeEnum.REFRESH);
+        nacosDataChangedListener.onAppAuthChanged(ImmutableList.of(appAuthData), DataEventTypeEnum.MYSELF);
+        nacosDataChangedListener.onAppAuthChanged(ImmutableList.of(appAuthData), DataEventTypeEnum.CREATE);
+
```
Comment (dengliming) [inline shenyu-admin/src/test/java/org/apache/shenyu/admin/listener/nacos/NacosDataChangedListenerTest.java:80]: verify(configService, times(4)).publishConfig(any(String.class), any(String.class), any(String.class));

**Review by dengliming — CHANGES_REQUESTED** [review]
Thanks for your contribution. I left some suggestions. Would you mind to take a look? : )

**Comment by renzhuyan** [comment]
> Thanks for your contribution. I left some suggestions. Would you mind to take a look? : )

Thank you for your suggestions. Please look again.

**Review by dengliming — APPROVED** [review]

**Pull request merged**